openmc/deplete/microxs.py

```diff
@@ -0,0 +1,229 @@
+            present in the inital composition.
+        dilute_initial : float
+            Initial atom density [atoms/cm^3] to add for nuclides that
+            are zero in initial condition to ensure they exist in the cross
+            section data. Only done for nuclides with reaction rates.
+        reactions : list of str, optional
+            Reaction names to tally
+        energy_bound : 2-tuple of float, optional
+            Bounds for the energy group.
+
+        Returns
+        -------
+        MicroXS
+            Cross section data in [b]
+
+        """
+        groups = EnergyGroups(energy_bounds)
+
+        # Set up the reaction tallies
+        original_tallies = model.tallies
+        tallies = Tallies()
+        xs = {}
+        reactions, diluted_materials = cls._add_dilute_nuclides(chain_file,
+                                                                model,
+                                                                dilute_initial)
+        diluted_materials.export_to_xml('diluted_materials.xml')
+        model.materials = diluted_materials
+
+        for rxn in reactions:
+            if rxn == 'fission':
+                xs[rxn] = FissionXS(domain=reaction_domain, groups=groups, by_nuclide=True)
+            else:
+                xs[rxn] = ArbitraryXS(rxn, domain=reaction_domain, groups=groups, by_nuclide=True)
+            tallies += xs[rxn].tallies.values()
+
+        model.tallies = tallies
+
+        # create temporary run
+        with tempfile.TemporaryDirectory() as temp_dir:
+            statepoint_path = model.run(cwd=temp_dir)
+
+            with StatePoint(statepoint_path) as sp:
+                for rxn in xs:
+                    xs[rxn].load_from_statepoint(sp)
+
+        # Build the DataFrame
+        micro_xs = cls()
+        for rxn in xs:
+            df = xs[rxn].get_pandas_dataframe(xs_type='micro')
+            df.index = df['nuclide']
+            df.drop(['nuclide', xs[rxn].domain_type, 'group in', 'std. dev.'], axis=1, inplace=True)
+            df.rename({'mean': rxn}, axis=1, inplace=True)
+            micro_xs = concat([micro_xs, df], axis=1)
+
+        # Revert to the original tallies
+        model.tallies = original_tallies
+
+        return micro_xs
+
+    @classmethod
+    def _add_dilute_nuclides(cls, chain_file, model, dilute_initial):
+        chain = Chain.from_xml(chain_file)
+        reactions = chain.reactions
+        cross_sections = _find_cross_sections(model)
+        nuclides_with_data = cls._get_nuclides_with_data(cross_sections)
+        burnable_nucs = [nuc.name for nuc in chain.nuclides
+                               if nuc.name in nuclides_with_data]
+        diluted_materials = Materials()
+        for material in model.materials:
+            if material.depletable:
+                nuc_densities = material.get_nuclide_atom_densities()
+                dilute_density = 1.E-24 * dilute_initial
```

Comment:
To match prevailing style throughout OpenMC codebase
```suggestion
                dilute_density = 1.e-24 * dilute_initial
```

Reply:
In that case there are a few other places this needs to be updated.